libs/SmartSync/src/com/salesforce/androidsdk/smartsync/util/SyncOptions.java

```diff
@@ -106,13 +126,23 @@ public JSONObject asJSON() throws JSONException {
 		JSONObject options = new JSONObject();
         if (mergeMode != null) options.put(MERGEMODE, mergeMode.name());
 		if (fieldlist != null) options.put(FIELDLIST, new JSONArray(fieldlist));
+		if (createFieldlist != null) options.put(CREATE_FIELDLIST, new JSONArray(createFieldlist));
+		if (updateFieldlist != null) options.put(UPDATE_FIELDLIST, new JSONArray(updateFieldlist));
 		return options;
 	}
```

Comment:
Seems like the create & update list does in other parts of code. Then maybe an if..else?